tests/functional/test_exclude_newer.py

```diff
@@ -0,0 +1,93 @@
+"""Tests for pip install --exclude-newer-than."""
+
+from __future__ import annotations
+
+import pytest
+
+from tests.lib import PipTestEnvironment, TestData
+
+
+class TestExcludeNewer:
+    """Test --exclude-newer-than functionality."""
+
+    def test_exclude_newer_than_invalid_date(
+        self, script: PipTestEnvironment, data: TestData
+    ) -> None:
+        """Test that --exclude-newer-than fails with invalid date format."""
+        result = script.pip(
+            "install",
+            "--no-index",
+            "-f",
+            data.packages,
+            "--exclude-newer-than=invalid-date",
+            "simple",
+            expect_error=True,
+        )
+
+        # Should fail with date parsing error
+        assert "invalid" in result.stderr.lower() or "error" in result.stderr.lower()
+
+    def test_exclude_newer_than_help_text(self, script: PipTestEnvironment) -> None:
+        """Test that --exclude-newer-than appears in help text."""
+        result = script.pip("install", "--help")
+        assert "--exclude-newer-than" in result.stdout
+        assert "datetime" in result.stdout
+
+    @pytest.mark.parametrize("command", ["install", "download", "wheel"])
+    def test_exclude_newer_than_available_in_commands(
+        self, script: PipTestEnvironment, command: str
+    ) -> None:
+        """Test that --exclude-newer-than is available in relevant commands."""
+        result = script.pip(command, "--help")
+        assert "--exclude-newer-than" in result.stdout
+
+    @pytest.mark.network
+    def test_exclude_newer_than_with_real_pypi(
+        self, script: PipTestEnvironment
+    ) -> None:
+        """Test exclude-newer functionality against real PyPI with upload times."""
+        # Use a small package with known old versions for testing
+        # requests 2.0.0 was released in 2013
```

Comment:
Could we pass `--no-deps` or use a package w/o dependencies? (e.g., six) Again, this is nit-picky, but I'd like to keep the tests as quick as reasonably possible. I know our test suite is by design not fast, but every little bit helps.